src/main/plugins/org.dita.base/xsl/preprocess/maplinkImpl.xsl

```diff
@@ -591,15 +586,10 @@ See the accompanying LICENSE file for applicable license.
         <!--figure out the linktext and desc-->
         <xsl:apply-templates select="*[contains(@class,' ditaot-d/ditaval-startprop ')]" mode="add-props-to-link"/>
         <xsl:if test="*[contains(@class, ' map/topicmeta ')]/*[dita-ot:matches-linktext-class(@class)]">
-          <!--Do not output linktext when The final output type is PDF or IDD
-            The target of the HREF is a local DITA file
-            The user has not specified locktitle to override the title -->
-          <xsl:if test="not(($FINALOUTPUTTYPE = 'PDF' or $FINALOUTPUTTYPE = 'IDD') and (not(@scope) or @scope = 'local') and (not(@format) or @format = 'dita') and (not(@locktitle) or @locktitle = 'no'))">
```

Comment:
`($FINALOUTPUTTYPE = 'PDF' or $FINALOUTPUTTYPE = 'IDD')` == `false()`

So this was `<xsl:if test="not(false() and X and Y and Z)">` ==> which gets `<xsl:if test="not(false())">` so again, always true